src/MetadataManager/metadataAPI.ts

```diff
@@ -0,0 +1,304 @@
+        const metadata = await Metadata.getMetadata(hash);
+        if (metadata === undefined) {
+            return;
+        }
+
+        // step 3. Check if the hash object has the deleted uri
+        const data = metadata[relativePath];
+        if(data) {
+            // step 4. If exists, deactivate (set 'is_deleted') that path.
+            metadata[relativePath]["isDeleted"] = true;
+            await Metadata.setMetadata(hash, metadata);
+
+            // step 5. Update pathToHash
+            pathToHash.deletePath(uri);
+        }
+    }
+
+    // deactivate all metadata under the folder
+    public static async disableMetadataUnderFolder(uri: vscode.Uri) {
+        // if it is a folder, deactivate all of its child files
+        const pathToHash = await PathToHash.getInstance();
+        for(let f of pathToHash.getFilesUnderFolder(uri)) {
+            if (!pathToHash.isFile(f)) {
+                await Metadata.disableMetadataUnderFolder(f);
+            } else if (Metadata.isValidFile(f)) {
+                await Metadata.disableMetadata(f);
+            }
+        }
+    }
+
+    public static async moveMetadata(oldUri: vscode.Uri, newUri: vscode.Uri) {
+        console.log('Metadata::moveMetadata()===========');
+        const oldRelativePath = vscode.workspace.asRelativePath(oldUri);
+        const newRelativePath = vscode.workspace.asRelativePath(newUri);
+        if(Metadata.isValidFile(oldUri) && !Metadata.isValidFile(newUri)) {
+            // when the file is renamed from a valid file name to a invalid file name
+            // ex. a.log > a.txt
+            await this.disableMetadata(oldUri);
+            return;
+        } else if(!Metadata.isValidFile(oldUri) || !Metadata.isValidFile(newUri)) {
+            return;
+        }
+        console.log(`moveMetadata:: old: ${oldRelativePath}, new: ${newRelativePath}`);
+
+        // 1. Get hash from pathToHash
+        const pathToHash = await PathToHash.getInstance();
+        const hash = pathToHash.getPathToHash(oldUri);
+        if (hash === undefined) {
+            return;
+        }
+
+        // 2. Get metadata from the old path
+        const metadata = await Metadata.getMetadata(hash);
+        // if there is no metadata, should we make a default metadata?
+        if (metadata === undefined) {
+            return;
+        }
+        const data = metadata[oldRelativePath];
+        if (data === undefined) {
+            return;
+        }
+        
+        // 3. Move metadata to the new path
+        delete metadata[oldRelativePath];
+        metadata[newRelativePath] = data;
+        await Metadata.setMetadata(hash, metadata);
+
+        // 4. Update pathToHash
+        await pathToHash.addPath(newUri);
+        pathToHash.deletePath(oldUri);
+    }
+
+    /**
+     * Move metadata of the files and folders under the fromUri folder to the toUri folder
+     */
+    public static async moveMetadataUnderFolder(fromUri: vscode.Uri, toUri: vscode.Uri) {
```

Comment:
(trivial) As it is vscode extension(microsoft), the term 'Directory' is commonly used instead of 'Folder'